Snowflake.Data.Tests/IntegrationTests/SFDbDataReaderIT.cs

```diff
@@ -1617,19 +1617,185 @@ public void TestTimestampNtz(string testValue, int scale)
+                selectCmd.CommandText = selectCommandText;
+                DbDataReader reader = selectCmd.ExecuteReader() as DbDataReader;
+
+                var enumerator = reader.GetEnumerator();
+                Assert.IsTrue(enumerator.MoveNext());
+                Assert.AreEqual(1, (enumerator.Current as DbDataRecord).GetInt64(0));
+                Assert.IsTrue(enumerator.MoveNext());
+                Assert.AreEqual(2, (enumerator.Current as DbDataRecord).GetInt64(0));
+                Assert.IsTrue(enumerator.MoveNext());
+                Assert.AreEqual(3, (enumerator.Current as DbDataRecord).GetInt64(0));
+
+                reader.Close();
+
+                cmd.CommandText = $"drop table if exists {TableName}";
+                var count = cmd.ExecuteNonQuery();
+                Assert.AreEqual(0, count);
+
+                CloseConnection(conn);
+            }
+        }
+
+        [Test]
+        public void TestGetEnumeratorShouldBeEmptyWhenNotRowsReturned()
+        {
+            using (var conn = CreateAndOpenConnection())
+            {
+                CreateOrReplaceTable(conn, TableName, new []{"cola NUMBER"});
+
+                IDbCommand cmd = conn.CreateCommand();
+
+                string insertCommand = $"insert into {TableName} values (1),(2),(3)";
+                cmd.CommandText = insertCommand;
+                cmd.ExecuteNonQuery();
+
+                string selectCommandText = $"select * from {TableName} WHERE cola > 10";
+                IDbCommand selectCmd = conn.CreateCommand();
+                selectCmd.CommandText = selectCommandText;
+                DbDataReader reader = selectCmd.ExecuteReader() as DbDataReader;
+
+                var enumerator = reader.GetEnumerator();
+                Assert.IsFalse(enumerator.MoveNext());
+                Assert.IsNull(enumerator.Current);
+
+                reader.Close();
+
+                cmd.CommandText = $"drop table if exists {TableName}";
+                var count = cmd.ExecuteNonQuery();
+                Assert.AreEqual(0, count);
+
+                CloseConnection(conn);
+            }
+        }
+
+        [Test]
+        public void TestGetEnumeratorWithCastMethod()
+        {
+            using (var conn = CreateAndOpenConnection())
+            {
+                CreateOrReplaceTable(conn, TableName, new []{"cola NUMBER"});
+
+                IDbCommand cmd = conn.CreateCommand();
+
+                string insertCommand = $"insert into {TableName} values (1),(2),(3)";
+                cmd.CommandText = insertCommand;
+                cmd.ExecuteNonQuery();
+
+                string selectCommandText = $"select * from {TableName}";
+                IDbCommand selectCmd = conn.CreateCommand();
+                selectCmd.CommandText = selectCommandText;
+                DbDataReader reader = selectCmd.ExecuteReader() as DbDataReader;
+
+                var x = reader.Cast<DbDataRecord>().ToList();
```

Comment:
could we find a better name for this variable than 'x'?

Reply:
Done, thank you!!